pkg/ring/lifecycler.go

```diff
@@ -478,6 +519,22 @@ func (i *Lifecycler) changeState(ctx context.Context, state IngesterState) error
 	return i.updateConsul(ctx)
 }
 
+func (i *Lifecycler) updateCounters(ringDesc *Desc) {
+	i.countersLock.Lock()
```

Comment:
Technically speaking, there is no need to keep the lock while counting healthy ingesters.